File: patterns/javascript-patterns/pattern45.js
// Pattern 45: ...

// 5*5*5
// 4*4*
// 3*3
// 2*
// 1

const n = 5;

for (let i = n; i >= 1; i--) {
    let row = '';

    for (let j = 1; j <= i; j++) {
        if (j % 2 === 0) {
            row += '*';
        } else {
            row += i;
        }
    }

    console.log(row);
}

File: patterns/python-patterns/pattern45.py
# Pattern 45: ...

# 1*3*5
# 1*3*
# 1*3
# 1*
# 1

n = 5

for i in range(n, 0, -1):
    for j in range(1, i + 1):
        if j % 2 == 0:
            print("*", end="")
        else:
            print(i, end="")
    print()
